salt/modules/mac_brew_pkg.py

```diff
@@ -31,7 +31,7 @@ def __virtual__():
     """
     if __grains__["os"] != "MacOS":
         return False, "brew module is macos specific"
-    if not salt.utils.path.which("brew"):
+    if not _homebrew_bin():
```

Comment:
@s0undt3ch this function calls `__salt__['cmd.run']` Can we rely on this module being loaded and availalbe in this virtual function?

Reply:
Good question. I'm not sure, but I'm leaning we can't guarantee.

Now, the question also is, do we have to use` __salt__` for that command?

Reply:
@arsa-dev since we are just ensuring `brew` is available and not actually running a brew command in the `__virtual__` func, couldn't we just keep this call in the `__virtual__` function to `salt.utils.path.which`?

Reply:
Hi @Ch3LL, first of all thanks you so much for your review and sorry for my delay, last week was a difficult one, I promise to be here more time this week so we can hopefully merge it 😊.

You are totally right, we shouldn't perform salt.run when we cannot fetch the os executable path with `salt.utils.path.which` and also there is no need to get the installation prefix when checking that `brew` is available since running `brew --prefix` should everytime return a valid output if we have the os executable path (except if brew installation is corrupt but this is something that I think it's out of scope).

I've rebased the branch with the latest master version and implemented a new `_homebrew_os_bin` func for fetching the os path with `which` without duplicating `salt.utils.path.which("brew")` instruction, used it at `__virutal__` func and also `_homebrew_bin` func that now comes with a safety extra check just in case the module is loaded and brew not available anymore, we are not calling cmd.run of `None --prefix`